### DISCUSSION TITLE
Hashing and styled system lib question

### DISCUSSION THREAD

**Discussion by omridevk** [chakra-ui/panda#2606] [Q&A]
I have two questions:
1. I am using panda-css and consuming my styled system as a package in my monorepo and I ship the buildinfo as well, is there any way to enable hashing only for the consumer of the styled system package? I don't want to configure the styled-system to have hashing by default.
2. Is there a way to dynamically set the hashing only for production?

**Comment by astahmer** [comment]
> I am using panda-css and consuming my styled system as a package in my monorepo and I ship the buildinfo as well, is there any way to enable hashing only for the consumer of the styled system package? I don't want to configure the styled-system to have hashing by default.

they'll have to use `hash: true` in their `panda.config`

> Is there a way to dynamically set the hashing only for production?

yes, you can use env vars like `process.env.NODE_ENV`

**Reply by omridevk** [reply]
ok so I can basically ship two versions of the buildinfo, one minified and one not?

**Comment by omridevk** [comment]
@astahmer 
I have a package that contains the `styled-system`, and I need to somehow tell it that only in dev mode to ship with no hashes, but I can't find a way to do it, as I need to generate the runtime with hash enabled in order for it to work, but then I have hashing all the time.
Does that makes sense?
I have three libs:
1. @omri/styled-system (contains the styled-system)
2. @omri/ui-design-system (consumes @omri/styled-system)
3. @omri/app - consumes both ui-design-system and @omri/styled-system

I need to make @omri/app produce hashed classes in prod, but no hashes in development, was not able to find a way to do this.

**Reply by omridevk** [reply]
I consume all these apps via NPM and ship the buildinfo (except for @omri/styled-system which nas no buildinfo)
